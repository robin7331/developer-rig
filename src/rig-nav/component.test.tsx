import { setupShallowTestWithStore, setupShallowTest } from '../tests/enzyme-util/shallow';
import { RigNav, RigNavComponent } from '.';
import { ExtensionViews, BroadcasterConfig, LiveConfig, Configurations, ProductManagement  } from '../constants/nav-items';
import { LoginButton } from '../login-button';
import { UserDropdown } from '../user-dropdown';
import { ManifestForTest } from '../tests/constants/extension';

describe('<RigNavComponent />', () => {
  const defaultGenerator = () => ({
    openConfigurationsHandler: jest.fn(),
    viewerHandler: jest.fn(),
    configHandler: jest.fn(),
    liveConfigHandler: jest.fn(),
    openProductManagementHandler: jest.fn(),
    selectedView: ExtensionViews,
    error: '',
    manifest: ManifestForTest,
    session: { login: 'test', profileImageUrl: 'test.png', authToken: 'test'},
  });
  const setupRenderer = setupShallowTest(RigNavComponent, defaultGenerator);
  const setupShallow = setupShallowTestWithStore(RigNav, defaultGenerator);

  it('renders correctly', () => {
    const { wrapper } = setupRenderer();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an error', () => {
    const { wrapper } = setupRenderer({
      error: 'test error',
    });

    expect(wrapper.find('.top-nav-error').text().trim()).toBe('test error');
  });

  it('correctly handles clicks on each tab', () => {
    const { wrapper } = setupRenderer();
    wrapper.find('a.top-nav-item').forEach((tab: any) => {
      tab.simulate('click');
    });
    expect(wrapper.instance().props.viewerHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.configHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.liveConfigHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.openProductManagementHandler).toHaveBeenCalled();
  });

  it('correct css classes are set when things are selected', () => {
    const { wrapper } = setupRenderer({
      selectedView: ExtensionViews,
    });
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);

    wrapper.setProps({
      selectedView: BroadcasterConfig,
    });
    wrapper.update();
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);

    wrapper.setProps({
      selectedView: LiveConfig,
    });
    wrapper.update();
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);

    wrapper.setProps({
      selectedView: ProductManagement,
    });
    wrapper.update();
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);
  });

  it('renders login button if no session', () => {
    const { wrapper } = setupRenderer({
      session: undefined,
    });
    expect(wrapper.find(LoginButton));
  });

  it('renders user dropdown if session exists', () => {
    const { wrapper } = setupRenderer({
      session: { login: 'test', profileImageUrl: 'test.png', authToken: 'test'},
    });
    expect(wrapper.find(UserDropdown));
  });

  it('disables product management tab when user is not logged in', () => {
    const { wrapper } = setupRenderer({
      session: undefined,
    });
    expect(wrapper.find('.top-nav-item__disabled')).toHaveLength(1);
  });

  it('disables product management tab when extension is not bits enabled', () => {
    const { wrapper } = setupRenderer({
      manifest: {
        ...ManifestForTest,
        bits_enabled: false,
      },
    });
    expect(wrapper.find('.top-nav-item__disabled')).toHaveLength(1);
  });
});
