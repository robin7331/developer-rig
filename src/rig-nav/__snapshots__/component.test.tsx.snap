// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RigNavComponent /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigNavComponent
    configHandler={[Function]}
    error=""
    liveConfigHandler={[Function]}
    manifest={
        Object {
            "anchor": "panel",
            "author_name": "test",
            "bits_enabled": true,
            "can_install": true,
            "config_url": "test",
            "description": "test",
            "eula_tos_url": "test.com",
            "icon_url": "test.com",
            "icon_urls": Object {},
            "id": "test",
            "installation_count": 0,
            "live_config_url": "test.com",
            "name": "test",
            "panel_height": 300,
            "privacy_policy_url": "test.com",
            "request_identity_link": false,
            "required_broadcaster_abilities": Array [
              "test",
            ],
            "screenshot_urls": Array [
              "test.png",
            ],
            "sku": "test",
            "state": "test",
            "summary": "test",
            "support_email": "test",
            "vendor_code": "test",
            "version": "0.0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelisted_config_urls": Array [],
            "whitelisted_panel_urls": Array [],
          }
    }
    openConfigurationsHandler={[Function]}
    openProductManagementHandler={[Function]}
    selectedView="Extension Views"
    session={
        Object {
            "authToken": "test",
            "login": "test",
            "profileImageUrl": "test.png",
          }
    }
    viewerHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="personal-bar"
>
          <Connect(MockApiDropdownComponent) />
          <div
                    className="personal-bar__configurations"
                    onClick={[Function]}
                    title="Configurations"
          >
                    <img
                              alt="Configurations"
                              height={24}
                              src="gear.svg"
                              width={24}
                    />
          </div>
          <div
                    className="personal-bar__ext-name"
          >
                    <span>
                              test
                    </span>
          </div>
          <div
                    className="top-nav-item__login"
          >
                    <Connect(UserDropdownComponent)
                              session={
                                        Object {
                                                  "authToken": "test",
                                                  "login": "test",
                                                  "profileImageUrl": "test.png",
                                                }
                              }
                    />
          </div>
</div>,
        <div
          className="top-nav__item-container"
>
          <a
                    className="offset top-nav-item top-nav-item__selected"
                    onClick={[Function]}
          >
                    Extension Views
          </a>
          <a
                    className="top-nav-item"
                    onClick={[Function]}
          >
                    Broadcaster Config
          </a>
          <a
                    className="top-nav-item"
                    onClick={[Function]}
          >
                    Live Config
          </a>
          <a
                    className="top-nav-item"
                    onClick={[Function]}
          >
                    Manage Bits Products
          </a>
</div>,
      ],
      "className": "top-nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(MockApiDropdownComponent) />,
            <div
              className="personal-bar__configurations"
              onClick={[Function]}
              title="Configurations"
>
              <img
                            alt="Configurations"
                            height={24}
                            src="gear.svg"
                            width={24}
              />
</div>,
            <div
              className="personal-bar__ext-name"
>
              <span>
                            test
              </span>
</div>,
            <div
              className="top-nav-item__login"
>
              <Connect(UserDropdownComponent)
                            session={
                                          Object {
                                                        "authToken": "test",
                                                        "login": "test",
                                                        "profileImageUrl": "test.png",
                                                      }
                            }
              />
</div>,
          ],
          "className": "personal-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Configurations"
                height={24}
                src="gear.svg"
                width={24}
/>,
              "className": "personal-bar__configurations",
              "onClick": [Function],
              "title": "Configurations",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Configurations",
                "height": 24,
                "src": "gear.svg",
                "width": 24,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                test
</span>,
              "className": "personal-bar__ext-name",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(UserDropdownComponent)
                session={
                                Object {
                                                "authToken": "test",
                                                "login": "test",
                                                "profileImageUrl": "test.png",
                                              }
                }
/>,
              "className": "top-nav-item__login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "session": Object {
                  "authToken": "test",
                  "login": "test",
                  "profileImageUrl": "test.png",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="offset top-nav-item top-nav-item__selected"
              onClick={[Function]}
>
              Extension Views
</a>,
            <a
              className="top-nav-item"
              onClick={[Function]}
>
              Broadcaster Config
</a>,
            <a
              className="top-nav-item"
              onClick={[Function]}
>
              Live Config
</a>,
            <a
              className="top-nav-item"
              onClick={[Function]}
>
              Manage Bits Products
</a>,
          ],
          "className": "top-nav__item-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Extension Views",
              "className": "offset top-nav-item top-nav-item__selected",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Extension Views",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Broadcaster Config",
              "className": "top-nav-item",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Broadcaster Config",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Live Config",
              "className": "top-nav-item",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Live Config",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Manage Bits Products",
              "className": "top-nav-item",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Manage Bits Products",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="personal-bar"
>
            <Connect(MockApiDropdownComponent) />
            <div
                        className="personal-bar__configurations"
                        onClick={[Function]}
                        title="Configurations"
            >
                        <img
                                    alt="Configurations"
                                    height={24}
                                    src="gear.svg"
                                    width={24}
                        />
            </div>
            <div
                        className="personal-bar__ext-name"
            >
                        <span>
                                    test
                        </span>
            </div>
            <div
                        className="top-nav-item__login"
            >
                        <Connect(UserDropdownComponent)
                                    session={
                                                Object {
                                                            "authToken": "test",
                                                            "login": "test",
                                                            "profileImageUrl": "test.png",
                                                          }
                                    }
                        />
            </div>
</div>,
          <div
            className="top-nav__item-container"
>
            <a
                        className="offset top-nav-item top-nav-item__selected"
                        onClick={[Function]}
            >
                        Extension Views
            </a>
            <a
                        className="top-nav-item"
                        onClick={[Function]}
            >
                        Broadcaster Config
            </a>
            <a
                        className="top-nav-item"
                        onClick={[Function]}
            >
                        Live Config
            </a>
            <a
                        className="top-nav-item"
                        onClick={[Function]}
            >
                        Manage Bits Products
            </a>
</div>,
        ],
        "className": "top-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(MockApiDropdownComponent) />,
              <div
                className="personal-bar__configurations"
                onClick={[Function]}
                title="Configurations"
>
                <img
                                alt="Configurations"
                                height={24}
                                src="gear.svg"
                                width={24}
                />
</div>,
              <div
                className="personal-bar__ext-name"
>
                <span>
                                test
                </span>
</div>,
              <div
                className="top-nav-item__login"
>
                <Connect(UserDropdownComponent)
                                session={
                                                Object {
                                                                "authToken": "test",
                                                                "login": "test",
                                                                "profileImageUrl": "test.png",
                                                              }
                                }
                />
</div>,
            ],
            "className": "personal-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Configurations"
                  height={24}
                  src="gear.svg"
                  width={24}
/>,
                "className": "personal-bar__configurations",
                "onClick": [Function],
                "title": "Configurations",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Configurations",
                  "height": 24,
                  "src": "gear.svg",
                  "width": 24,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  test
</span>,
                "className": "personal-bar__ext-name",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                },
                "ref": null,
                "rendered": "test",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(UserDropdownComponent)
                  session={
                                    Object {
                                                      "authToken": "test",
                                                      "login": "test",
                                                      "profileImageUrl": "test.png",
                                                    }
                  }
/>,
                "className": "top-nav-item__login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "session": Object {
                    "authToken": "test",
                    "login": "test",
                    "profileImageUrl": "test.png",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="offset top-nav-item top-nav-item__selected"
                onClick={[Function]}
>
                Extension Views
</a>,
              <a
                className="top-nav-item"
                onClick={[Function]}
>
                Broadcaster Config
</a>,
              <a
                className="top-nav-item"
                onClick={[Function]}
>
                Live Config
</a>,
              <a
                className="top-nav-item"
                onClick={[Function]}
>
                Manage Bits Products
</a>,
            ],
            "className": "top-nav__item-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Extension Views",
                "className": "offset top-nav-item top-nav-item__selected",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Extension Views",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Broadcaster Config",
                "className": "top-nav-item",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Broadcaster Config",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Live Config",
                "className": "top-nav-item",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Live Config",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Manage Bits Products",
                "className": "top-nav-item",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Manage Bits Products",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
